Can't install Erlang-24 due to openssl issues

Getting the following when installing erlang 24 via `asdf install erlang latest` on MacOS Big Sur 11.3.1:

```checking for OpenSSL header in /usr/local/opt/openssl@1.1... yes
checking for OpenSSL in /usr/local/opt/openssl@1.1... configure: error: static linking against crypto library disabled by user, but no dynamic library found in /usr/local/opt/openssl@1.1
ERROR: /Users/adonis/.asdf/plugins/erlang/kerl-home/builds/asdf_24.0/otp_src_24.0/lib/crypto/configure failed!
```

That’s with the --with-ssl option specified or not. I did run `asdf plugin update --all` prior to running the install erlang latest command above.

On a slightly older version of Big Sur (11.2.3), but getting a similar error. 

```
checking for OpenSSL header in /usr/local/opt/openssl@1.1... yes
checking for OpenSSL in /usr/local/opt/openssl@1.1... configure: error: neither static nor dynamic crypto library found in /usr/local/opt/openssl@1.1
ERROR: /Users/USERNAME/.asdf/plugins/erlang/kerl-home/builds/asdf_24.0/otp_src_24.0/lib/crypto/configure failed!
```

Tried reinstalling openssl via `brew reinstall openssl` and also manually specifying the `export KERL_CONFIGURE_OPTIONS="--with-ssl=$(brew --prefix openssl)"` flag

Thanks for workaround @ronyclau! Seems to work for `asdf`, but has anyone has success with it applied to `kerl` directly? I'm trying

```
kerl build 24.0 24.0
```
then in the other terminal
```
printf '/if test "x$DED_LD" = "x"; then\ni\nunset DED_LD\n.\n\nw\nq\n' | ed .kerl/builds/24.0/otp_src_24.0/lib/crypto/configure
```
I see the "patch successful" lines, but the build fails with the same `configure: error: neither static nor dynamic crypto library` as before.

--- a/kerl
+++ b/kerl
@@ -788,34 +788,22 @@ do_normal_build() {
 }
 
 _flags() {
-    # We need to munge the LD and DED flags for clang 9/10 shipped with
-    # High Sierra (macOS 10.13) and Mojave (macOS 10.14) and quite
-    # probably for Catalina (macOS 10.15)
+    # We used to munge the LD and DED flags for clang 9/10 shipped with
+    # High Sierra (macOS 10.13), Mojave (macOS 10.14) and Catalina
+    # (macOS 10.15)
+    #
+    # As of OTP 20.1 that is (apparently) no longer necessary and
+    # in OTP 24 breaks stuff. See thread and comment here:
+    # https://github.com/erlang/otp/issues/4821#issuecomment-845914942
     case "$KERL_SYSTEM" in
         Darwin)
-            osver=$(uname -r)
-            case "$osver" in
-                # TODO: Remove this in a future kerl release, probably
-                # after Catalina (10.15)
-                19*|18*|17*)
-                    # Make sure we don't overwrite values that someone who
-                    # knows better than us set.
-                    if [ -z "$DED_LD" ]; then
-                        DED_LD='clang'
-                    fi
-                    if [ -z "$CC" ]; then
-                        CC='clang'
-                    fi
-                    if [ -z "$DED_LDFLAGS" ]; then
-                        host=$(./erts/autoconf/config.guess)
-                        DED_LDFLAGS="-m64 -bundle -bundle_loader ${ERL_TOP}/bin/$host/beam.smp"
-                    fi
-                    CFLAGS="$CFLAGS" DED_LD="$DED_LD" CC="$CC" DED_LDFLAGS="$DED_LDFLAGS" "$@"
-                    ;;
-                *)
-                    CFLAGS="$CFLAGS" "$@"
-                    ;;
-            esac
+            # Make sure we don't overwrite stuff that someone who
+            # knows better than us set.
+            if [ -z "$CC" ]; then
+                CC='clang'
+            fi
+
+            CFLAGS="$CFLAGS" CC="$CC" "$@"
             ;;
         *)
             CFLAGS="$CFLAGS" "$@"

